pinot-core/src/main/java/org/apache/pinot/core/startree/v2/builder/BaseSingleTreeBuilder.java

```diff
@@ -150,7 +151,11 @@
       // Ignore the column for COUNT aggregation function
       if (_valueAggregators[index].getAggregationType() != AggregationFunctionType.COUNT) {
         String column = functionColumnPair.getColumn();
-        _metricDataTypes[index] = schema.getFieldSpecFor(column).getDataType();
+        FieldSpec fieldSpec = schema.getFieldSpecFor(column);
+        if (fieldSpec == null) {
```

Comment:
You can write if block into a single line.
```
Precondition.checkNotNull(fieldSpec, "Invalid column: " + column);
```

Reply:
I would suggest using checkState instead:
```suggestion
        Preconditions.checkState(fieldSpec != null, "Column: '%s' does not exist in the schema", column);
```